File: SpaceChallenge_Java/app/src/main/java/com/example/spacechallenge_java/U2.java
package com.example.spacechallenge_java;

public class U2 extends Rocket{
    public U2() {
        super(120, 18000, 29000, 0.04, 0.08);
        setTotalWeight(getSpaceshipWeight());
    }

    @Override
    public boolean launch() {
        double random = Math.random();
        return getChanceOfLaunchExplosion() * (getTotalWeight() - getSpaceshipWeight()) / (getMaxWeight() - getSpaceshipWeight()) <= random;
    }
    @Override
    public boolean land() {
        double random = Math.random();
        return getChanceOfLandingCrash() * (getTotalWeight() - getSpaceshipWeight()) / (getMaxWeight() - getSpaceshipWeight()) <= random;
    }
}

File: SpaceChallenge_Kotlin/app/src/main/java/com/savelievoleksandr/spacechallenge_kotlin/U2.kt
package com.savelievoleksandr.spacechallenge_kotlin

class U2 : Rocket(120, 18000, 29000, 0.04, 0.08) {

    override fun launch(): Boolean {
        val random = Math.random()
        return chanceOfLaunchExplosion * (totalWeight - spaceshipWeight) / (maxWeight - spaceshipWeight) <= random
    }

    override fun land(): Boolean {
        val random = Math.random()
        return chanceOfLandingCrash * (totalWeight - spaceshipWeight) / (maxWeight - spaceshipWeight) <= random
    }
}